client/scripts/models/CommunityInfo.ts

```diff
@@ -37,12 +37,44 @@ class CommunityInfo {
     );
   }
 
+  public async updateCommunityData(
+    name: string,
+    description: string,
+    privacyEnabled: boolean,
+    invitesEnabled: boolean
+  ) {
+    try {
+      await $.post(`${app.serverUrl()}/updateCommunity`, {
+        'id': app.activeCommunityId(),
+        'name': name,
+        'description': description,
+        'privacy': privacyEnabled,
+        'invites': invitesEnabled,
+        'jwt': app.login.jwt,
+      }).then((r) => {
+        const updatedCommunity: CommunityInfo = r.result;
+        this.name = updatedCommunity.name;
+        this.description = updatedCommunity.description;
+        this.privacyEnabled = updatedCommunity.privacyEnabled;
+        this.invitesEnabled = updatedCommunity.invitesEnabled;
+      });
+    } catch (err) {
+      console.log('Failed to update Community Data');
+      throw new Error((err.responseJSON && err.responseJSON.error)
+        ? err.responseJSON.error
+        : 'Failed to update Community Data');
+    }
+  }
+
   public async updateFeaturedTags(tags: string[]) {
     try {
       await $.post(`${app.serverUrl()}/updateCommunity`, {
         'id': app.activeCommunityId(),
         'featured_tags[]': tags,
         'jwt': app.login.jwt
+      }).then((result) => {
```

Comment:
this is bad style, why await into a promise?

Reply:
fixed!